src/python/gem5/simulate/simulator.py

```diff
@@ -360,6 +384,46 @@ def function_generator(func: Callable):
 
         self._checkpoint_path = checkpoint_path
 
+    def set_id(self, id: int) -> None:
+        self._id = id
+
+    def get_id(self) -> int:
+        """
+        Returns the ID of the simulation. This is particularly useful when
+        running multiple simulations in parallel. The ID can be unique and
+        descriptive of the simulation. If set via the contructor or the
+        `set_id` function, this ID is return. If not set a hash of the
+        simulated system configuration is used.
+
+        :raises Exception: An exception is raised if the id is not set via
+                           the constructor or via the `set_id` function, and
+                           the simulation is yet to be instantiated.
+        """
+
+        if hasattr(self, "_id") and self._id:
+            return self._id
+
+        if not self._instantiated:
+            raise Exception(
+                "If not set via the Simulator constructor, or via `set_id`, "
+                "the simulator/simulation ID cannot be obtained "
+                "prior to initialization as it is based, partially, "
+                "on a hash of the instiaite simulated system."
+            )
+
+        return self.hash().hex().lower()
+
+    def set_max_ticks(self, max_tick: int) -> None:
```

Comment:
I like this function. Can you add a docstring that specifies whether this is a delta or absolute tick (I assume absolute, but it's important to make that clear).